server/src/main/java/org/elasticsearch/threadpool/DefaultBuiltInExecutorBuilders.java

```diff
@@ -145,6 +146,12 @@ public Map<String, ExecutorBuilder> getBuilders(Settings settings, int allocated
                 false
             )
         );
+        if (ThreadPoolMergeScheduler.USE_THREAD_POOL_MERGE_SCHEDULER_SETTING.get(settings)) {
+            result.put(
+                ThreadPool.Names.MERGE,
+                new ScalingExecutorBuilder(ThreadPool.Names.MERGE, 1, allocatedProcessors, TimeValue.timeValueMinutes(5), false)
```

Comment:
I wonder if we should use `true` here for reject after shutdown. Otherwise the task just vanishes.

Reply:
Yes, the thread pool executor now rejects merge tasks after shutdown (this has been addressed across multiple PRs, and there are also tests covering this, e.g. `ThreadPoolMergeExecutorServiceTests#testNewMergeTaskIsAbortedWhenThreadPoolIsShutdown`).